src/poly/commitment.rs

```diff
@@ -188,6 +190,48 @@ impl<C: CurveAffine> Params<C> {
     pub fn get_g(&self) -> Vec<C> {
         self.g.clone()
     }
+
+    /// Writes params to a buffer.
+    pub fn write<W: io::Write>(&self, writer: &mut W) -> io::Result<()> {
+        writer.write_all(&self.k.to_le_bytes())?;
+        writer.write_all(&self.n.to_le_bytes())?;
```

Comment:
I don't think we technically need both, since we assume `n = 2^k` in various parts of the codebase, but it's fine to include both.

Reply:
I disagree; I think we should just write one of `k` or `n`, and calculate the other when reading. (Otherwise we should have an assertion that they're consistent when reading, which is no easier than calculating.)

Reply:
We now write only `k`, and calculate `n` when reading.